sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -135,6 +137,95 @@ def test_encoding(categories, unknown_value, global_random_seed, smooth, target_
     assert_allclose(X_test_transform, expected_X_test_transform)
 
 
+@pytest.mark.parametrize("smooth", [5.0, "auto"])
+def test_encoding_multiclass(global_random_seed, smooth):
+    """Check encoding for multiclass targets."""
+    rng = np.random.RandomState(global_random_seed)
+
+    n_samples = 80
+    n_features = 2
+    feat_1 = rng.randint(low=0, high=2, size=n_samples)
+    feat_2 = rng.randint(low=0, high=3, size=n_samples)
+    X_train = np.column_stack((feat_1, feat_2))
+    categories = [[0, 1], [0, 1, 2]]
+
+    n_classes = 3
+    y_train = rng.randint(low=0, high=n_classes, size=n_samples)
+    y_train_enc = LabelBinarizer().fit_transform(y_train)
+
+    n_splits = 3
+    random_state = 0
+    cv = StratifiedKFold(n_splits=n_splits, random_state=random_state, shuffle=True)
+
+    # Manually compute encodings for cv splits to validate `fit_transform`
+    expected_X_fit_transform = np.empty(
+        (X_train.shape[0], X_train.shape[1] * n_classes),
+        dtype=np.float64,
+    )
+    for f_idx, cats in enumerate(categories):
+        for c_idx in range(n_classes):
+            for train_idx, test_idx in cv.split(X_train, y_train):
+                y_class = y_train_enc[:, c_idx]
+                X_, y_ = X_train[train_idx, f_idx], y_class[train_idx]
+                current_encoding = _encode_target(X_, y_, len(cats), smooth)
+                # f_idx:   0, 0, 0, 1, 1, 1
+                # c_idx:   0, 1, 2, 0, 1, 2
+                # exp_idx: 0, 1, 2, 3, 4, 5
+                exp_idx = c_idx + (f_idx * n_classes)
+                expected_X_fit_transform[test_idx, exp_idx] = current_encoding[
+                    X_train[test_idx, f_idx]
+                ]
+
+    target_encoder = TargetEncoder(
+        smooth=smooth,
+        categories=categories,
+        cv=n_splits,
+        random_state=random_state,
+    )
+    X_fit_transform = target_encoder.fit_transform(X_train, y_train)
+
+    assert target_encoder.target_type_ == "multiclass"
+    assert_allclose(X_fit_transform, expected_X_fit_transform)
+
+    # Manually compute encoding to validate `transform`
+    expected_encodings = []
+    for f_idx, cats in enumerate(categories):
+        for c_idx in range(n_classes):
+            y_class = y_train_enc[:, c_idx]
+            current_encoding = _encode_target(
+                X_train[:, f_idx], y_class, len(cats), smooth
+            )
+            expected_encodings.append(current_encoding)
+
+    assert len(target_encoder.encodings_) == n_features * n_classes
+    for i in range(n_features * n_classes):
+        assert_allclose(target_encoder.encodings_[i], expected_encodings[i])
+
```

Comment:
Please also insert an assertion about the content of `target_encoder.classes_`.

Reply:
Not also added assertion about content of `target_encoder.classes_` to `test_encoding` to test the binary case